schemas/file.yml

```diff
@@ -88,4 +88,14 @@
   - name: ctime
     level: extended
     type: date
-    description: Last time file metadata changed.
+    description: Last time the file attributes or metadata changed.
+
+  - name: created
+    level: extended
+    type: date
+    description: File creation time.
+
+  - name: accessed
+    level: extended
+    type: date
+    description: Last time file content accessed.
```

Comment:
What if a file was renamed? Does this count? Does this correlate with `touch foo.txt`?

Reply:
Hmmm, I guess we could detail which actions typically update which timestamp, on Windows and Linux families. But I'm sure there's edge cases.

Perhaps a more straightforward approach (and easier to maintain) would be to state which timestamps this maps to, for each platform family?

Reply:
Is it different between platforms? Don't know much about the windows ones.

Reply:
I would say this should align with the last access time that the different platforms report natively? I assume that's what users would expect.

Linux and macOS track access times by default and show them in the output of e.g. `stat` (`stat -x` on macOS). Windows [no longer tracks](https://superuser.com/a/251265/66061) access time by default for performance reasons, but it can be enabled.

Maybe the description should say `Last time file was accessed.`?

Reply:
I've submitted a small tweak to the description, my initial sentence wasn't great :-)

But I feel like getting into the detail of what happens on renames, `touch` & the like will not be helpful, and will be somewhat out of place.

I'd vote to leave as is, and update if there seems to be confusion / questions about this.

Reply:
I think the part I'm more concerned is how this competes / conflicts for example with `ctime`. It's super easy to add new fields but will be very hard to remove fields again. So if we add a field, we should be very clear what it's used for and how.

Reply:
Sorry I hadn't noticed this last comment, when I responded on the PR itself.

I'm not sure what you meant by "this" in "how this competes".

In #441, we're clarifying a quirk of `ctime` vs `mtime`, [see this](https://github.com/elastic/ecs/pull/441/files?file-filters%5B%5D=.yml#diff-776162149979f3e81d54d5bf9b843027R115).

The cryptic `crtime` often confuses people vs `ctime`. This is why instead we're introducing `created`, as the field for `crtime`. So I think this is step 1 in clarifying these timestamps.

Step 2 of clarifying these timestamps will be getting rid of the already existing `ctime` and `mtime`. But as already discussed in [my comment a few hours ago](https://github.com/elastic/ecs/pull/445#issuecomment-488311438), this should be done separately. I want do have good support for deprecations.

Since we're improving the field descriptions over in #441 and introducing new ones here, let's pull them all together, to have a complete overview of the proposed changes:

- **mtime**: Last time the file content was modified.
- **ctime**: Last time the file attributes or metadata changed.
       Note that changes to the file content will update `mtime`. This implies
      `ctime` will be adjusted at the same time, since `mtime` is an attribute
      of the file.
- **created**: File creation time.
       Note that not all filesystems store the creation time.
- **accessed**: Last time the file was accessed.
       Note that not all filesystems keep track of access time.

Is there anything missing or unclear here?

If you're questioning whether they're worth adding at all, I do think so. Both are fundamental pieces of information with no overlap with the 2 timestamps we already have. The fact that not all filesystems / configurations record creation time or last access time for performance reasons is not a reason for not adding them, IMO. When they're available, I think they're useful.